doc/config.rst

```diff
@@ -1633,6 +1633,17 @@ These options influence LaTeX output.
         .. versionadded:: 1.2
      ``'fontenc'``
         "fontenc" package inclusion, default ``'\\usepackage[T1]{fontenc}'``.
+     ``'amsmath'``
```

Comment:
I will remove this in next major release (I haven't done that yet, but I agree what you commented at https://github.com/sphinx-doc/sphinx/commit/d6572259d66cfb9da967f45fb2d671d29c5d5683#commitcomment-17423029)
So, personally, it is better to keep it undocumented.